### PR TITLE
Kill S3ExternalIdImageStore

### PR DESCRIPTION
**Pull request by leogrim**
@ryanpbrewster

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/model/UserPicture.scala [hunk 1]
```diff
@@ -35,6 +36,13 @@ case class UserPicture(
 
 object UserPicture {
   def generateNewFilename: String = RandomStringUtils.randomAlphanumeric(5)
+
+  def toS3Key(size: String, userId: ExternalId[User], pic: String): String = s"users/$userId/pics/$size/$pic"
```
Comment (ryanpbrewster) [inline kifi-backend/modules/common/app/com/keepit/model/UserPicture.scala]: `${userId.id}`

**Comment by andrewconner** [comment]
:us: :fr: :bomb:

File: kifi-backend/modules/common/app/com/keepit/social/BasicUser.scala [hunk 2]
```diff
@@ -38,6 +39,7 @@ case class BasicUser(
     pictureName: String,
     username: Username) extends BasicUserFields {
   def path: Path = Path(username.value)
+  def avatarPath: ImagePath = UserPicture.toImagePath(Some(200), externalId, pictureName)
```
Comment (ryanpbrewster) [inline kifi-backend/modules/common/app/com/keepit/social/BasicUser.scala]: I think `avatar` is currently used exclusively for organizations. How about `picturePath`?

**Comment by ryanpbrewster** [comment]
This is nice, I was confused about why we had so many distinct layers of abstraction on this.

File: kifi-backend/modules/common/app/com/keepit/model/UserPicture.scala (hunk 1 above)
Comment (leogrim) [inline kifi-backend/modules/common/app/com/keepit/model/UserPicture.scala]: Good catch :+1:

**Pull request merged**